File: 14-friend-groups/go/main.go
package main

import (
	"bufio"
	"encoding/json"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type input struct {
	N       int     `json:"n"`
	Unions  [][]int `json:"unions"`
	Queries [][]int `json:"queries"`
}

func solve(n int, unions, queries [][]int) []int {
	_, _, _ = n, unions, queries
	return nil
}

func main() {
	var in input
	if err := json.NewDecoder(os.Stdin).Decode(&in); err != nil {
		fmt.Fprintln(os.Stderr, err)
		os.Exit(1)
	}
	out := solve(in.N, in.Unions, in.Queries)
	parts := make([]string, len(out))
	for i, v := range out {
		parts[i] = strconv.Itoa(v)
	}
	w := bufio.NewWriter(os.Stdout)
	defer w.Flush()
	fmt.Fprintln(w, strings.Join(parts, " "))
}

File: 14-friend-groups/rust/src/main.rs
use std::io::{self, Read};
use union_find::{solve, Input};

fn main() {
    let mut buf = String::new();
    io::stdin().read_to_string(&mut buf).unwrap();
    let inp: Input = serde_json::from_str(&buf).unwrap();
    let out = solve(inp.n, &inp.unions, &inp.queries);
    let parts: Vec<String> = out.iter().map(|v| v.to_string()).collect();
    println!("{}", parts.join(" "));
}
